backend/infrahub/events/proposed_change_action.py

```diff
@@ -0,0 +1,112 @@
+    reviewer_account_name: str = Field(..., description="The name of the user who reviewed the proposed change")
+    reviewer_decision: str = Field(..., description="The decision made by the reviewer")
+
+    def get_resource(self) -> dict[str, str]:
+        return {
+            **super().get_resource(),
+            "infrahub.proposed_change.reviewer_account_id": self.reviewer_account_id,
+            "infrahub.proposed_change.reviewer_account_name": self.reviewer_account_name,
+            "infrahub.proposed_change.reviewer_decision": self.reviewer_decision,
+        }
+
+
+class ProposedChangeReviewRevokedEvent(ProposedChangeEvent):
+    reviewer_account_id: str = Field(..., description="The ID of the user who reviewed the proposed change")
+    reviewer_account_name: str = Field(..., description="The name of the user who reviewed the proposed change")
+    reviewer_former_decision: str = Field(..., description="The former decision made by the reviewer")
+
+    def get_resource(self) -> dict[str, str]:
+        return {
+            **super().get_resource(),
+            "infrahub.proposed_change.reviewer_account_id": self.reviewer_account_id,
+            "infrahub.proposed_change.reviewer_account_name": self.reviewer_account_name,
+            "infrahub.proposed_change.reviewer_former_decision": self.reviewer_former_decision,
+        }
+
+
+class ProposedChangeMergedEvent(ProposedChangeEvent):
+    """Event generated when a proposed change has been merged"""
+
+    event_name: ClassVar[str] = f"{EVENT_NAMESPACE}.proposed_change.merged"
+
+    merged_by_account_id: str = Field(..., description="The ID of the user who merged the proposed change")
+    merged_by_account_name: str = Field(..., description="The name of the user who merged the proposed change")
+
+    def get_resource(self) -> dict[str, str]:
+        return {
+            **super().get_resource(),
+            "infrahub.proposed_change.merged_by_account_id": self.merged_by_account_id,
+            "infrahub.proposed_change.merged_by_account_name": self.merged_by_account_name,
+        }
+
+
+class ProposedChangeReviewRequestedEvent(ProposedChangeEvent):
+    """Event generated when a proposed change has been flagged for review"""
+
+    event_name: ClassVar[str] = f"{EVENT_NAMESPACE}.proposed_change.review_requested"
+
+    requested_by_account_id: str = Field(
+        ..., description="The ID of the user who requested the proposed change to be reviewed"
+    )
+    requested_by_account_name: str = Field(
+        ..., description="The name of the user who requested the proposed change to be reviewed"
+    )
+
+    def get_resource(self) -> dict[str, str]:
+        return {
+            **super().get_resource(),
+            "infrahub.proposed_change.review_requested_by_account_id": self.requested_by_account_id,
+            "infrahub.proposed_change.review_requested_by_account_name": self.requested_by_account_name,
+        }
```

Comment:
Looking at this again I realize that we typically use get_related() for the event specific data points. Would perhaps be nice in terms of consistency. For some events like the group or node mutation ones it's required as we can have multiple entries with the same name. In this case I'm not sure it's really needed. What do you think?

Reply:
I haven't thought about this, but this is a good idea. If we leverage `get_related()`, would it also simplify the GraphQL exposed objects?

Reply:
I would say that it's fairly similar, an example is the artifacts with the method `_return_artifact_event()`